### ISSUE TITLE
Crash when using GifTextureView in ListView

### ISSUE THREAD

**Issue by huyduongtu** [koral--/android-gif-drawable#240]
I'm using GifTextureView in ListView item.
The code segment in Adapter is following:
```
@Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        if (convertView == null) {
            convertView = LayoutInflater.from(mContext).inflate(R.layout.texture_item, parent, false);
            holder = new ViewHolder();
            holder.imageView = (GifTextureView) convertView.findViewById(R.id.gif);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        holder.imageView.setInputSource(new InputSource.FileSource(imgSources[position]));

        return convertView;
    }

    public static class ViewHolder {
        GifTextureView imageView;
    }
```
When I press back away off this activity. It crashes or show dialog ANR.
```
01-07 14:17:55.495 13166-13365/huy.com.gifresearch E/AndroidRuntime: FATAL EXCEPTION: GifRenderThread
Process: huy.com.gifresearch, PID: 13166
java.lang.RuntimeException: Display loop poll failed Interrupted system call
    at huy.com.libgifdrawable.GifInfoHandle.bindSurface(Native Method)
    at huy.com.libgifdrawable.GifInfoHandle.bindSurface(GifInfoHandle.java:134)
    at huy.com.libgifdrawable.GifTextureView$RenderThread.run(GifTextureView.java:218)
```

**Comment by koral--** [comment]
That exception occurs because process has been interrupted by clicking OK/Force Close button in crash/ANR dialog. That another crash/ANR is not necessarily caused by this library.
Basically waiting for the moment of frame rendering is interrupted because app process is being killed.
Only thing I can do is to ignore interruption and exit from rendering loop gracefully. You need to find a reason of ANR/another uncaught exception.

**Comment by huyduongtu** [comment]
I don't know why. My listview's item is GifTextureView. When I press Back to exit the activity. It always crashes or shows ANR dialog.
What should I do to avoid this?

**Comment by koral--** [comment]
Could you post complete logcat?

**Comment by koral--** [comment]
OK, I was able to reproduce ANR. I'll fix it ASAP.

**Comment by huyduongtu** [comment]
This is all in my adapter.
```
public class GifTextureViewAdapter extends BaseAdapter {
    private Context mContext;
    private String[] imgSources;
    
    public GifTextureViewAdapter(Context context) {
        mContext = context;
        imgSources = Images.getImgSources();
    }

    @Override
    public int getCount() {
        return imgSources.length;
    }

    @Override
    public Object getItem(int position) {
        return imgSources[position];
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        if (convertView == null) {
            convertView = LayoutInflater.from(mContext).inflate(R.layout.texture_item, parent, false);
            holder = new ViewHolder();
            holder.imageView = (GifTextureView) convertView.findViewById(R.id.gif);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        holder.imageView.setInputSource(new InputSource.FileSource(imgSources[position]));

        return convertView;
    }

    public static class ViewHolder {
        GifTextureView imageView;
    }
}
```
The log crash is:
```
FATAL EXCEPTION: GifRenderThread
Process: huy.com.gifresearch, PID: 17708
java.lang.RuntimeException: Display loop poll failed Interrupted system call
        at huy.com.libgifdrawable.GifInfoHandle.bindSurface(Native Method)
        at huy.com.libgifdrawable.GifInfoHandle.bindSurface(GifInfoHandle.java:134)
        at huy.com.libgifdrawable.GifTextureView$RenderThread.run(GifTextureView.java:218)
```

It always crashes or show ARN dialog

**Comment by koral--** [comment]
Fixed in 1.1.12-SNAPSHOT.
